File: Cisco/7BestSightseeingPair.py
class Solution(object):
    def maxScoreSightseeingPair(self, values):
        """
        :type values: List[int]
        :rtype: int
        """
        cur_max = values[0]
        max_score = -1

        for el in values[1:]:
            cur_max -= 1
            if cur_max + el > max_score:
                max_score = cur_max + el
            cur_max = max(cur_max, el)

        return max_score

File: Cisco/LinkedInList/C++/7BestSightSeeingPair.cpp
#include <vector>
#include <span>

using namespace std;

class Solution
{
public:
    int maxScoreSightseeingPair(vector<int> &values)
    {
        int N = values.size();
        int currMax = values[0], maxScore = -1;
        for (int i = 1; i < N; ++i)
        {
            currMax -= 1;
            maxScore = max(maxScore, currMax + values[i]);
            currMax = max(currMax, values[i]);
        }

        return maxScore;
    }
};